pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -114,9 +114,13 @@ func (s *statsSyncLoad) SendLoadRequests(sc *stmtctx.StatementContext, neededHis
 			select {
 			case s.StatsLoad.NeededItemsCh <- task:
 				metrics.SyncLoadDedupCounter.Inc()
-				result, ok := <-task.ResultCh
-				intest.Assert(ok, "task.ResultCh cannot be closed")
-				return result, nil
+				select {
+				case <-timer.C:
```

Comment:
It seems weird that both L118 and L124 are listening to `timer.C`.

Reply:
Another part is that this is within singleflight, so it should exit promptly. Otherwise, it will keep accumulating, and others cannot trigger a sync load.

Reply:
I gave it some thought and decided to add a timer to determine whether the task is genuinely stuck. Additionally, I reduced the number of retries since sync load tasks are often triggered by queries, making excessive retries less meaningful.